src/openrct2/core/FileStream.h

```diff
@@ -53,87 +53,9 @@ namespace OpenRCT2
         uint64_t _fileSize = 0;
 
     public:
-        FileStream(const std::string& path, int32_t fileMode)
-            : FileStream(path.c_str(), fileMode)
-        {
-        }
-
-        FileStream(const utf8* path, int32_t fileMode)
-        {
-            const char* mode;
-            switch (fileMode)
-            {
-                case FILE_MODE_OPEN:
-                    mode = "rb";
-                    _canRead = true;
-                    _canWrite = false;
-                    break;
-                case FILE_MODE_WRITE:
-                    mode = "w+b";
-                    _canRead = true;
-                    _canWrite = true;
-                    break;
-                case FILE_MODE_APPEND:
-                    mode = "a";
-                    _canRead = false;
-                    _canWrite = true;
-                    break;
-                default:
-                    throw;
-            }
-
-            // Make sure the directory exists before writing to a file inside it
-            if (_canWrite)
-            {
-                std::string directory = Path::GetDirectory(path);
-                if (!Path::DirectoryExists(directory))
-                {
-                    Path::CreateDirectory(directory);
-                }
-            }
-
-#ifdef _WIN32
-            auto pathW = String::ToWideChar(path);
-            auto modeW = String::ToWideChar(mode);
-            _file = _wfopen(pathW.c_str(), modeW.c_str());
-#else
-            if (fileMode == FILE_MODE_OPEN)
-            {
-                struct stat fileStat;
-                // Only allow regular files to be opened as its possible to open directories.
-                if (stat(path, &fileStat) == 0 && S_ISREG(fileStat.st_mode))
-                {
-                    _file = fopen(path, mode);
-                }
-            }
-            else
-            {
-                _file = fopen(path, mode);
-            }
-#endif
-            if (_file == nullptr)
-            {
-                throw IOException(String::StdFormat("Unable to open '%s'", path));
-            }
-
-            Seek(0, STREAM_SEEK_END);
-            _fileSize = GetPosition();
-            Seek(0, STREAM_SEEK_BEGIN);
-
-            _ownsFilePtr = true;
-        }
-
-        ~FileStream() override
-        {
-            if (!_disposed)
-            {
-                _disposed = true;
-                if (_ownsFilePtr)
-                {
-                    fclose(_file);
-                }
-            }
-        }
+        FileStream(const std::string& path, int32_t fileMode);
+        FileStream(const utf8* path, int32_t fileMode);
+        ~FileStream() override;
 
         bool CanRead() const override
```

Comment:
why not move more of the bellow over so we can move the #ifdef macro stuff into the cpp file.

Reply:
Not sure why I skipped these, maybe because they were small. I've moved them to the .cpp files now, as well as the includes and macros at the top.